NewForce-Cohort-5/tabloidfullstack-actual-spring
# Hb add tag to post tkt17

# Description

Given the author is viewing Post details
When they select the Manage Tags option
Then they should be presented with a selection of Tags options to associate with the Post
Given the author has selected the Tags they wish to associate with a Post
When they click the Save button
Then the association should be saved to the database
And the user should be directed to the Post details page
And the selected Tags should be displayed on the Post details page

In the backend, I have added the update method to the Post Repository  and Post Controller.   

The front end, I added a Manage Tag File and updated Post Details


Fixes # (issue)

## Type of change

Please delete options that are not relevant.


# Testing Instructions

1. After verifying that you have our repo cloned down into a local repository, `git fetch -a` in the main branch to retrieve the named `hb-addTagToPost-Tkt17` that I have pushed up
2. In Terminal `git checkout `hb-addTagToPost-Tkt17`  to switch to my branch and in the terminal type `Start TabloidMVC.sln` to open in Visual Studios 2022.
3. To get to VS code, type `cd Tabloid `and then `cd Client` and type` code .`
4. If you have not seeded your database with this data, seed your data from SQLQuery link :
5. You will need to create the table first and then add the seed data. Choose files in the following order:   

-    [01_Tabloid_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-actual-spring/blob/main/SQL/01_Tabloid_Create_DB.sql)
-    [02_Tabloid_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-actual-spring/blob/main/SQL/02_Tabloid_Seed_Data.sql)


6. Now that you have your seed data, we can test.
7. Click the `solid green debugger play button` and you will arrive at the swagger page. We don’t need to do anything here. This just confirms the server is running. 
8. In your terminal, you should be in the client folder where `Package.json` is located. You want to type `npm start` to start your react app.  
9. You will arrive at a login page where you will use the email login as [foo@bar.com](mailto:foo@bar.com) and password test.
10. Next go to `Post` in the menu. 
11. Please test at least 2 times by switching between menu items links and logging in and out
12. Click on a post to go to details and click on manage tag. 
13. Here you will add a tag.
14. You will be sent back to post details where you can see the tag. 



# Checklist:
